src/main/java/de/unijena/cheminf/fragment/fingerprint/performanceTest/PerformanceTest.java

```diff
@@ -0,0 +1,419 @@
+    private ArrayList<HashMap<String, Integer>> moleculeFragmentList = new ArrayList<>(this.INITIAL_CAPACITY_VALUE);
+    //</editor-fold>
+    //
+    //<editor-fold defaultstate="collapsed" desc="Constructor">
+    /**
+     * Constructor.
+     * All files listed below as parameters must be located in the same directory as the application JAR file.
+     * In the constructor all necessary files for storing the performance test are created.
+     * There are 3 different files to document the result of the performance snapshot.
+     * The results during the generation of the bit and count fingerprint are stored once
+     * individually and once together.
+     * Subsequently, the fingerprints are created, and the results are documented in the corresponding files.
+     *
+     * @param anArgs the command line arguments, anArgs[0] and anArgs[1] must be the names of the text files to load.
+     *               The text files must be located in the same directory as the application's JAR file. The command
+     *               line anArgs[3] must be the name of the path, where the performance test results are
+     *               stored. The command line anArgs[2] must be the number of the molecules in process.
+     * @throws IOException is thrown if the constructor is unable to open a text file for logging occurred exceptions.
+     */
+    public PerformanceTest(String[] anArgs) throws IOException {
+        if(anArgs.length !=4) {
+            throw new IllegalArgumentException("Four arguments (three file names and the number of the molecules in process) are required.");
+        }
+        String tmpFilePath =  (new File("").getAbsoluteFile().getAbsolutePath()) + File.separator;
+        /*Set up exception log file*/
+        this.workingPath = (new File(anArgs[2]).getAbsoluteFile().getAbsolutePath()) + File.separator;
+        LocalDateTime tmpDateTime = LocalDateTime.now();
+        String tmpProcessingTime = tmpDateTime.format(DateTimeFormatter.ofPattern("uuuu_MM_dd_HH_mm"));
+        new File(this.workingPath + "/Results").mkdirs();
+        File tmpExceptionsLogFile = new File(this.workingPath + "/Results/"
+                + PerformanceTest.EXCEPTIONS_LOG_FILE_NAME);
+        FileWriter tmpExceptionsLogFileWriter = new FileWriter(tmpExceptionsLogFile, true);
+        this.exceptionsPrintWriter = new PrintWriter(tmpExceptionsLogFileWriter);
+        this.exceptionsPrintWriter.println("#########################################################################");
+        this.exceptionsPrintWriter.println("Processing Time: " + tmpProcessingTime);
+        this.exceptionsPrintWriter.println();
+        this.exceptionsPrintWriter.flush();
+        try {
+            // Load fragment files
+            this.moleculeFile = new File(tmpFilePath + anArgs[0]);
+            this.fragmentFile = new File(tmpFilePath + anArgs[1]);
+            FileInputStream tmpFragmentFileInputStream;
+            FileInputStream tmpMoleculeFileInputStream;
```

Comment:
IntelliJ says these two input streams are unused

Reply:
Are they just for testing here? Then please add an according comment

Reply:
More importantly, they are not closed